### PR TITLE
Add login enabled field to header settings.

### PR DESCRIPTION
**Pull request by migh**
This PR creates a admin -> style -> header option for display or hide login.

## Testing instructions

1. For a local site go to Admin -> Style -> Header
2. You should see a new **Login** option.
3. You can choose show or hide and save.
4. The site should display or omit the login item accordingly.

image

image

## Pivotal Tracker

[Task](https://www.pivotaltracker.com/story/show/168435693)

### CODE REVIEW

File: app/models/site_setting.rb [hunk 1]
```diff
@@ -239,6 +243,10 @@ def self.create_style_settings site
       site.site_settings.find_or_initialize_by(name: 'footer-links-color', value: '\'white\'', position: 31)
     end
 
+    unless site.site_settings.exists?(name: 'header_login_enabled')
+      site.site_settings.find_or_initialize_by(name: 'header_login_enabled', value: 'false', position: 32)
```
Comment (clementprdhomme) [inline app/models/site_setting.rb]: I think that by default, the login should be shown. Only a handful of projects will disable it.

Comment (migh) [reply app/models/site_setting.rb]: Sure, sounds fair. I'll wait for Jesus opinion, mine is that is just the way I wrote that condition, for making the changes.

Comment (jesusgonzalezrivera) [reply app/models/site_setting.rb]: The faster solution is to change the condition but I think that the most consistent solution is to add a new migration or rake task which creates this new setting for all sites

File: app/models/site_page.rb [hunk 2]
```diff
@@ -176,6 +176,11 @@ def attributes
     attrs
   end
 
+  def header_login_enabled?
+    flag = site.site_settings.try(:header_login_enabled, site_id).try(:value)
```
Comment (jesusgonzalezrivera) [inline app/models/site_page.rb]: This line is calling to the `header_login_enabled` that you have defined on `SiteSetting` model. So we can change it by this:
```
SiteSetting.header_login_enabled(site_id)&.value
```

Comment (migh) [inline app/models/site_page.rb]:
```suggestion
    flag = SiteSetting.header_login_enabled(site_id)&.value
```

**Pull request merged**